dashboard/client/src/common/CollapsibleSection.tsx

```diff
@@ -106,16 +93,63 @@ export const CollapsibleSection = forwardRef<
           </Typography>
           {icon}
         </Box>
-        {(finalExpanded || (keepRendered && rendered)) && (
-          <div
-            className={classNames(classes.body, {
-              [classes.bodyHidden]: !finalExpanded,
-            })}
-          >
-            {children}
-          </div>
-        )}
+        <HideableBlock visible={finalExpanded} keepRendered={keepRendered}>
+          {children}
+        </HideableBlock>
       </div>
     );
   },
 );
+
+const useHideableBlockStyles = makeStyles((theme) =>
+  createStyles({
+    body: {
+      marginTop: theme.spacing(1),
+    },
+    bodyHidden: {
+      display: "none",
+    },
+  }),
+);
+
+type HideableBlockProps = PropsWithChildren<
+  {
+    visible?: boolean;
+    /**
+     * An optimization to not avoid re-rendering the contents of the collapsible section.
+     * When enabled, we will keep the content around when collapsing but hide it via css.
+     */
+    keepRendered?: boolean;
+  } & ClassNameProps
+>;
+
+/**
+ * Component that can be hidden depending on a passed in prop. Supports an optimization
```

Comment:
Could you move the second line of the docstring to L140 or L146? I think that comment would be better colocated where the logic is cause this function took me a bit to wrap my head around.

Also shouldn't visible be a required arg?

Reply:
good catch.